lib/class-wp-rest-block-types-controller.php

```diff
@@ -0,0 +1,326 @@
+		$block_type = WP_Block_Type_Registry::get_instance()->get_registered( $block_name );
+
+		if ( empty( $block_type ) ) {
+			return new WP_Error( 'rest_block_type_invalid', __( 'Invalid block type.', 'gutenberg' ), array( 'status' => 404 ) );
+		}
+
+		$data = $this->prepare_item_for_response( $block_type, $request );
+
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Prepares a block type object for serialization.
+	 *
+	 * @param stdClass        $block_type block type data.
+	 * @param WP_REST_Request $request    Full details about the request.
+	 *
+	 * @return WP_REST_Response block type data.
+	 */
+	public function prepare_item_for_response( $block_type, $request ) {
+
+		$fields = $this->get_fields_for_response( $request );
+		$data   = array();
+		if ( in_array( 'name', $fields, true ) ) {
+			$data['name'] = $block_type->name;
+		}
+
+		if ( in_array( 'attributes', $fields, true ) ) {
+			$data['attributes'] = $block_type->get_attributes();
+		}
+
+		if ( in_array( 'is_dynamic', $fields, true ) ) {
+			$data['is_dynamic'] = $block_type->is_dynamic();
+		}
+
+		$extra_fields = array( 'editor_script', 'script', 'editor_style', 'style' );
+		foreach ( $extra_fields as $extra_field ) {
+			if ( in_array( $extra_field, $fields, true ) ) {
+				$data[ $extra_field ] = (string) $block_type->$extra_field;
+			}
+		}
+
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
+		$data    = $this->add_additional_fields_to_object( $data, $request );
+		$data    = $this->filter_response_by_context( $data, $context );
+
+		$response = rest_ensure_response( $data );
+
+		$response->add_links(
+			array(
+				'collection'              => array(
+					'href' => rest_url( sprintf( '%s/%s', $this->namespace, $this->rest_base ) ),
+				),
+				'https://api.w.org/items' => array(
+					'href' => rest_url( sprintf( '%s/%s/%s', $this->namespace, $this->rest_base, $block_type->name ) ),
+				),
+			)
+		);
+
+		/**
+		 * Filters a block type returned from the REST API.
+		 *
+		 * Allows modification of the block type data right before it is returned.
+		 *
+		 * @param WP_REST_Response $response   The response object.
+		 * @param object           $block_type The original block type object.
+		 * @param WP_REST_Request  $request    Request used to generate the response.
+		 */
+		return apply_filters( 'rest_prepare_block_type', $response, $block_type, $request );
+	}
+
+	/**
+	 * Retrieves the block type' schema, conforming to JSON Schema.
+	 *
+	 * @return array Item schema data.
+	 */
+	public function get_item_schema() {
+		if ( $this->schema ) {
+			return $this->add_additional_fields_schema( $this->schema );
+		}
+
+		$schema = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'block_type',
+			'type'       => 'object',
+			'properties' => array(
+				'name'          => array(
+					'description' => __( 'The title for the block type.', 'gutenberg' ),
```

Comment:
It could be misleading if/when the actual `title` field is included.

In the [RFC document](https://github.com/WordPress/gutenberg/blob/master/docs/rfc/block-registration.md#name), it's described as "a unique string that identifies a block". Maybe we can adapt that here.

```suggestion
					'description' => __( 'Unique name identifying the block type.', 'gutenberg' ),
```